src/action/finder/request.rs

```diff
@@ -0,0 +1,1405 @@
+// Copyright 2020 Google LLC
+//
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+//! Defines an internal type for client side file finder action and provides
+//! a function converting proto format of the request
+//! `rrg_proto::FileFinderArgs` to the internal format.
+
+use crate::session::{
+    parse_enum, time_from_micros, ParseError, ProtoEnum, RegexParseError,
+};
+use regex::Regex;
+use rrg_proto::{
+    FileFinderAccessTimeCondition, FileFinderAction, FileFinderArgs,
+    FileFinderCondition, FileFinderContentsLiteralMatchCondition,
+    FileFinderContentsRegexMatchCondition, FileFinderDownloadActionOptions,
+    FileFinderExtFlagsCondition, FileFinderHashActionOptions,
+    FileFinderInodeChangeTimeCondition, FileFinderModificationTimeCondition,
+    FileFinderSizeCondition, FileFinderStatActionOptions,
+};
+use std::convert::TryFrom;
+
+type HashActionOversizedFilePolicy =
+    rrg_proto::file_finder_hash_action_options::OversizedFilePolicy;
+type DownloadActionOversizedFilePolicy =
+    rrg_proto::file_finder_download_action_options::OversizedFilePolicy;
+type RegexMatchMode =
+    rrg_proto::file_finder_contents_regex_match_condition::Mode;
+type LiteralMatchMode =
+    rrg_proto::file_finder_contents_literal_match_condition::Mode;
+type ActionType = rrg_proto::file_finder_action::Action;
+type ConditionType = rrg_proto::file_finder_condition::Type;
+type XDevMode = rrg_proto::file_finder_args::XDev;
+type PathType = rrg_proto::path_spec::PathType;
+
+#[derive(Debug)]
+pub struct Request {
+    pub paths: Vec<String>,
+    pub path_type: PathType,
```

Comment:
Is this used?

Reply:
It is used e.g. for setting the type to "Registry".

Reply:
Removed `path_type` from code.